cmd/gke-gcloud-auth-plugin/cred/cred.go

```diff
@@ -231,9 +246,15 @@ func (p *plugin) getCachedGcloudAccessToken() (string, *metav1.Time, error) {
 	if err != nil {
 		return "", nil, fmt.Errorf("error retrieving starting config: %w", err)
 	}
+
 	// If current context is not the same as what the cached access token was
 	// generated for, then consider the current access token invalid.
-	if c.CurrentContext != startingConfig.CurrentContext {
+	currentContext := startingConfig.CurrentContext
+	if providerContext := p.tokenProvider.context(); providerContext != "" {
+		currentContext += "_" + providerContext
+	}
```

Comment:
If the "cluster" change in the kubeconfig, the startingConfig.CurrentContext will also change. CurrentContext refers to the user+cluster. Can you please explain how this helps?

Reply:
That is good scenario feedback, but, we don't want to support manually running the plugin(as we have no known use case), as that will open a can of worms :(

Reply:
Removed, thanks